config/config_helpers.go

```diff
@@ -62,22 +63,37 @@ type EnvVar struct {
 
 // Create an EvalContext for the HCL2 parser. We can define functions and variables in this context that the HCL2 parser
 // will make available to the Terragrunt configuration during parsing.
-func CreateTerragruntEvalContext(include *IncludeConfig, terragruntOptions *options.TerragruntOptions) *hcl.EvalContext {
+func CreateTerragruntEvalContext(filename string, include *IncludeConfig, terragruntOptions *options.TerragruntOptions) *hcl.EvalContext {
+	tfscope := tflang.Scope{
+		BaseDir: filepath.Dir(filename),
```

Comment:
Using the directory the current terragrunt file is in seemed reasonable here, but suggestions are welcome.

Reply:
Perhaps the working dir? That's available in `terragruntOptions.WorkingDir`, IIRC.

Reply:
Is `terragruntOptions.WorkingDir` the user's CWD?  If so, `BaseDir` is used as the base to resolve files from so it might be confusing if file resolution changes based on where you run terragrunt from.

Reply:
`terragruntOptions.WorkingDir` is the directory where Terragrunt is running `terraform` commands. E.g., if you have a `source = "<MODULE>"` configuration in Terragrunt, then `<MODULE>` will be checked out into the `WorkingDir`, which is `.terragrunt-cache/xxx/yyy`. If you don't have a `source = "..."`, then `WorkingDir` is whatever folder you're running Terragrunt in.

Which of the helpers use `BaseDir`? And what would a user expect it to be?

Reply:
These are the functions that use it:
```
			"file":             funcs.MakeFileFunc(s.BaseDir, false),
			"fileexists":       funcs.MakeFileExistsFunc(s.BaseDir),
			"filebase64":       funcs.MakeFileFunc(s.BaseDir, true),
			"filebase64sha256": funcs.MakeFileBase64Sha256Func(s.BaseDir),
			"filebase64sha512": funcs.MakeFileBase64Sha512Func(s.BaseDir),
			"filemd5":          funcs.MakeFileMd5Func(s.BaseDir),
			"filesha1":         funcs.MakeFileSha1Func(s.BaseDir),
			"filesha256":       funcs.MakeFileSha256Func(s.BaseDir),
			"filesha512":       funcs.MakeFileSha512Func(s.BaseDir),
```

Maybe it's best to just use `/` and require absolute paths, since `get_terragrunt_dir` and `get_parent_terragrunt_dir` exist?  The docs don't seem to mention what happens to relative paths: https://www.terraform.io/docs/configuration/functions/file.html

Reply:
Also I'm not positive what the order of things is, but do the files get copied to `terragruntOptions.WorkingDir` before the terragrunt files are evaluated?

Reply:
Ah, OK, thanks for the context. Yes, `terragrunt.hcl` is parsed and all the functions are executed as the first step, before any files are copied. Given this info, and the helpers you listed, using the parent dir of the `terragrunt.hcl` file via the `filename` parameter, as you have it, makes sense to me. We should just make sure to document this!